clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3222,8 +3242,11 @@ void Sema::MarkDevice(void) {
 
   for (Decl *D : syclDeviceDecls()) {
     if (auto SYCLKernel = dyn_cast<FunctionDecl>(D)) {
-      llvm::SmallPtrSet<FunctionDecl *, 10> VisitedSet;
-      Marker.CollectKernelSet(SYCLKernel, SYCLKernel, VisitedSet);
+
```

Comment:
syclDeviceDecls includes SYCL_EXTERNAL functions, not only kernels, correct?
NIT:
I suggest to replace "Kernel" substring in the added code with something more accurate - TopLevelFunction, ExternalFunction, ExportedFunction,...

Reply:
> syclDeviceDecls includes SYCL_EXTERNAL functions, not only kernels, correct?

Yes, there is the corresponding code at the beginning of the `MarkDevice` function that handles that.

> NIT:
> I suggest to replace "Kernel" substring in the added code with something more accurate - TopLevelFunction, ExternalFunction, ExportedFunction,...

I agree. Here we collect a set of functions called from the SYCL/ESIMD kernel. So, maybe `collectSyclKernelFuncs` or `collectSyclKernelCallees`? And ESIMD counterparts, like `collectEsimdKernelCallees`.

Reply:
My point is that the roots of the graph the collection works on include not only kernels, but also SYCL_EXTERNAL functions. So `Kernel` in the name is somewhat misleading.

Reply:
Ah, I see. The maybe `collectSyclCalleesSet`/`collectEsimdCalleesSet` ?

Reply:
The `SYCLKernel` variable name is itself wrong since it's not only kernels but also SYCL_EXTERNAL functions. :)

Reply:
collectSyclCallGraphNodes/collectESIMDCallGraphNodes?

Reply:
Ok, I like your variant more.